File: easy/2784_check_if_array_is_good/Solution.java
class Solution {
  public boolean isGood(int[] nums) {
    final int n = nums.length;
    final int[] numsCounter = new int[n];

    for (int i = 0; i < n; ++i) {
      if (nums[i] >= n) {
        return false;
      }

      numsCounter[nums[i]]++;

      if (numsCounter[nums[i]] > 1 && nums[i] != n - 1) {
        return false;
      }
    }

    return numsCounter[n - 1] == 2;
  }
}

File: easy/2784_check_if_array_is_good/solution.cpp
#include <map>
#include <vector>

class Solution {
public:
  bool isGood(std::vector<int> &nums) {
    std::map<int, int> numsCounter;
    for (const int &num : nums)
      ++numsCounter[num];

    const int maxNum = numsCounter.rbegin()->first;

    if (numsCounter[maxNum] != 2)
      return false;

    for (int num = 1; num < maxNum; ++num) {
      if (numsCounter[num] != 1)
        return false;
    }

    return true;
  }
};
